### PR TITLE
Purchase Order

### PR DESCRIPTION
**Pull request by vivekshekar**
@abhikchakraborty 

I have created purchase order api, but I cannot test this. As I cannot create purchase order on my local. I have created an issue with respect this on sqcrm issue,

https://github.com/sQcrm/sqcrm/issues/244

Please check the same and let me know.

### CODE REVIEW

File: controllers/purchaseorder.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+						   
+						return autoCallback(null, purchaseorder);
+					});
+				}]
+			},
+			function(err, results) {
+				if (err) return next(err);
+				var purchaseorders = {},
+					purchseorderPrefix = results.getPurchaseOrderPrifix;
+
```
Comment (abhikchakraborty) [inline controllers/purchaseorder.js:141]: `purchseorderPrefix` there is a typo and its used later as `purchaseorderPrefix` line:163

JS hint shows the errors as 
```
163 |                                if (purchaseorderPrefix && purchaseorderPrefix.settingData && k === 'po_number') {
                                             ^ 'purchaseorderPrefix' is not defined.
    163 |                                if (purchaseorderPrefix && purchaseorderPrefix.settingData && k === 'po_number') {
                                                                    ^ 'purchaseorderPrefix' is not defined.
    164 |                                    v= purchaseorderPrefix.settingData+''+v;
                                                ^ 'purchaseorderPrefix' is not defined.

```

File: controllers/purchaseorder.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+		*          defaultValue: 50
+		*/
+		getAll: function(req, res, next) {
+			var apiBase = req.protocol + '://' + req.get('host') + apiNamespace,
+				apiEndpoint = apiBase + '/' + resourceType + '/',
+				whereClause,
+				page,
+				limit;
+
+			crmPrivileges.userWhereCondition(req, 16, 'PurchaseOrder', 'purchase_order_custom_fld', true, function(err, whereCond) {
```
Comment (abhikchakraborty) [inline controllers/purchaseorder.js]: The 2nd param in the above function is the table name https://github.com/sQcrm/sqcrm-node-api/blob/pruchase_order/utils/crm_privileges.js#L285 
You are sending the module name, the table name is `purchase_order`

File: controllers/purchaseorder.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+									v= purchaseorderPrefix.settingData+''+v;
+									purchaseorder[k] = v;
+								}
+							});
+							return modifiedCallBack();
+						},
+						// set the line items which is one-to-many relation with quote
+						setLineItems: function(modifiedCallBack) {
+							app.models.lineitems 
+							.find({recordid:purchase_order.id,moduleId:16})
```
Comment (abhikchakraborty) [inline controllers/purchaseorder.js]: `urchase_order.id` is undefined, and the loop is going as 
```
async.forEachOf(results.getRecords, function(purchaseorder, key, eachCallBack) {
```
So it should be as

```
.find({recordid:purchaseorder.id,moduleId:16})
```
Also all the variables, functions etc should be `camelCase` not `snake case`

**Review by abhikchakraborty — CHANGES_REQUESTED** [review]
@vivekshekar I had reviewed the task and left some comment. Please check.

File: controllers/purchaseorder.js (hunk 1 above)
Comment (vivekshekar) [reply controllers/purchaseorder.js:141]: @abhikchakraborty fixed the issues

**Pull request merged**